src/Authoring/WinRT.StubExe/WinRT.StubExe.cpp

```diff
@@ -0,0 +1,52 @@
+#include "pch.h"
+#include <windows.h>
+
+#pragma comment(lib, "shell32.lib")
+
+typedef int (*__managed__Main)(int, wchar_t*[]);
+
+int APIENTRY wWinMain(
+    _In_ HINSTANCE hInstance,
+    _In_opt_ HINSTANCE hPrevInstance,
+    _In_ LPWSTR    lpCmdLine,
+    _In_ int       nCmdShow)
+{
+    wchar_t fileName[MAX_PATH];
+
+    // Get the path of the current .exe file (it will be renamed to match the app name)
+    if (!GetModuleFileNameW(NULL, fileName, MAX_PATH))
+    {
+        return GetLastError();
+    }
+
+    int fileNameLength = lstrlenW(fileName);
+
+    // Replace the extension (.exe -> .dll)
+    memcpy(
+        /* _Dst */ &fileName[fileNameLength - 3 /* "exe" */],
+        /* _Src */ L"dll",
+        /* _Size */ sizeof(wchar_t) * 3 /* "dll" */);
+
+    // Load the .dll for the app
+    HMODULE hModule = LoadLibraryW(fileName);
+
+    if (!hModule)
+    {
+        return GetLastError();
+    }
+
+    // Get the custom main from the native .dll (hardcoded to "__managed__Main")
+    FARPROC pEntryPoint = GetProcAddress(hModule, "__managed__Main");
+
+    if (!pEntryPoint)
+    {
+        return GetLastError();
+    }
+
+    // We need to parse the arguments to get back 'argc', 'argv' for the managed entry point
+    int argc;
+    LPWSTR* argv = CommandLineToArgvW(lpCmdLine, &argc);
```

Comment:
This is broken legacy API with number of corner-case bugs: https://github.com/dotnet/runtime/issues/77644

Is there no way to convince C++ compiler to build WinRT app with regular C main?

Reply:
Sergio is using the GUI subsystem here, which uses WinMain rather than C main.

Reply:
Oh, TIL. Technically speaking this stub .exe doesn't have to be a WinRT app whatsoever (it literally just needs to be some native executable jumping into that custom main, that's it, it doesn't do anything WinRT-related on its own, nor does it have to know about it even). I just created a C++ project as my understanding was that compiling flat C required a bit of trickery compared to vanilla C++, but if there's a simple way to do that, absolutely, we could do that! I was mostly just prototyping things here so far 🙂

Reply:
Apparently you can change the `/ENTRY` linker flag to `mainCRTStartup` to use C main.

https://learn.microsoft.com/en-us/cpp/build/reference/entry-entry-point-symbol?view=msvc-170

Reply:
Ooh good find! Done in 6d4caaeb4ad781d9ba122794e8adf2222ab7e515 😄